src/scripts/create_hunter_csv.py

```diff
@@ -0,0 +1,331 @@
+            ' op/sec')
+        for col_name in raw_hunter_metrics_df.columns:
+            if col_name not in ['totalOps', 'opRate']:
+                raw_hunter_metrics_df[col_name] = raw_hunter_metrics_df[col_name].str.rstrip(
+                    ' ms')
+        # Get date from the json path (regardless of its positional index) and add time for compatibility with hunter.
+        list_of_items_from_json_path = json_paths[0].split(os.sep)
+
+        date_val = ''
+        for item_in_json in list_of_items_from_json_path:
+            matched_pattern = re.search(DATE_DIR_REGEX_PATTERN, item_in_json)
+            if matched_pattern:
+                date_val = matched_pattern.group()
+
+        # As at 11pm UTC
+        date_val_w_time = f"{date_val.replace('_', '-')}{' '}{LWT_TEST_RUN_EXEC_TIME}{' +0000'}"
+
+        # Get the Git shas for the Cassandra and fallout-tests repos for auditability
+        cassandra_git_short_hash = get_git_sha_for_cassandra(date_val)
+        fallout_tests_git_short_hash = get_git_sha_for_fallout_tests(date_val)
+
+        combined_read_write_df = create_hunter_df(raw_hunter_metrics_df)
+
+        hunter_df_full = add_cols_to_metrics_df(
+            date_val_w_time,
+            cassandra_git_short_hash,
+            fallout_tests_git_short_hash,
+            combined_read_write_df
+        )
+        return hunter_df_full
+
+
+def get_hunter_df_w_test_type(json_paths: List[str]) -> Tuple[pd.DataFrame, str]:
+    """
+    Get the dataframe to feed to hunter with performance results and the corresponding test type (e.g., 100/1000/10000
+    partitions and either 'fixed' or 'rated').
+
+    Args:
+        json_paths: List[str]
+                    A list of json paths with performance results and related metrics.
+
+    Returns:
+            A tuple with the dataframe with performance results and the corresponding test type.
+    """
+    hunter_df_out = generate_hunter_df(json_paths)
+
+    if len(json_paths) == 0 or hunter_df_out.empty:
+        logging.error(
+            "Either the 'json_paths' or the hunter dataframe is empty; "
+            "thus, empty dataframe and string are being returned."
+        )
+        return pd.DataFrame(), ''
+
+    # Get the test type based on a tuple of supported tests (regardless of their positional index)
+    list_of_items_from_json_path = json_paths[0].split(os.sep)
+    test_type_str = ''
+    for item_in_json in list_of_items_from_json_path:
+        if item_in_json.startswith(TUPLE_SUPPORTED_TESTS):
+            test_type_str = item_in_json
+
+    if test_type_str != '' and not hunter_df_out.empty:
+        return hunter_df_out, test_type_str
+
+
+if __name__ == '__main__':
+    # Get sorted list of dates from output folders with performance tests
+    nightly_result_dates = os.listdir(NIGHTLY_RESULTS_DIR)
+    nightly_result_dates.sort()
+
+    # Set to False if running this for the first time, then re-run and set to True.
+    is_case_prospective = PROSPECTIVE_MODE
```

Comment:
In general, it's too high of a bar to expect users to change the code to alter the behaviour of an app/script. In 6 months someone is going to run this script and, I suspect, not read the code :) This should really be a command-line argument with the default being prospective.

Also, the explanation of prospective/retrospective should be in the README.

But these are minor suggestions and I think we should just create a follow up GH issue to record these two points and address it later. It doesn't need to hold up merging this PR.

Reply:
I have added the explanation on README at [99de7fd](https://github.com/datastax/fallout-tests/pull/5/commits/99de7fd8c3562ee2cc17c780c309c98e6f257c64) and as for the command line argument, will create a GH issue for this.